### PR TITLE
Navigation fixes

### PR DESCRIPTION
**Pull request by antsgar**
**Description**

- Fixes missing entities errors in Finances subsections.
- Fixes tasks cards not sizing gracefully.
- Fixes buggy styling on crop detail page.

Jira link:

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by kathyavini — COMMENTED** [review]
These look like good fixes to me!

I'm still not seeing Actual Revenues on the first load though (visiting directly after login). Expenses is perfectly fixed for me.

File: packages/webapp/src/containers/Finances/ActualRevenue/index.jsx [hunk 1]
```diff
@@ -49,9 +51,9 @@ export default function ActualRevenue({ history, match }) {
   );
 
   useEffect(() => {
```
Comment (kathyavini) [inline packages/webapp/src/containers/Finances/ActualRevenue/index.jsx:53]: An inelegant but working solution would be to just call `fetchAllData()` here 😬

File: packages/webapp/src/containers/Finances/ActualRevenue/index.jsx [hunk 2]
```diff
@@ -49,9 +51,9 @@ export default function ActualRevenue({ history, match }) {
...
-    if (!allRevenueTypes?.length) {
-      dispatch(getRevenueTypes());
-    }
+    dispatch(getRevenueTypes());
```
Comment (kathyavini) [inline packages/webapp/src/containers/Finances/ActualRevenue/index.jsx:54]: An inelegant but working solution seems to be just calling `fetchAllData()` here instead 😬

Comment (antsgar) [reply packages/webapp/src/containers/Finances/ActualRevenue/index.jsx]: So I have no idea why, but the getSales action wasn't actually working -- setting it up with `createAction` as some of the others seems to have done the trick

**Review by kathyavini — COMMENTED** [review]
Strange about `ADD_SALE` but it works well now!

Can you remind me where we landed regarding this error where the `sale_id` isn't being populated in the URL? It looks familiar (was it mentioned on one of the previous nav PRs?) but I can't fully remember. This is on beta/integration too, not just this branch:

https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/dda5f32d-7186-4f7c-b1a4-6cbc86da13e8

Also, as a minor cosmetic point, I'm wondering if the scroll bars should be hidden in all cases on the new menu even when the content overflows due to a smaller screen height (it happens in the video above when Finances is expanded). The default scroll bar looks a little wonky to me here and breaks estimated revenue onto two lines 🤔

**Comment by antsgar** [comment]
@kathyavini I (hopefully) fixed the redirects using this syntax https://github.com/remix-run/react-router/pull/5209, can you re-test please? 🙏 re the scrollbars, I think it's best not to override the browser's default behavior for accessibility/UX, since the bars are a visual cue that the content is scrolling. They look a bit different for me though in Chrome, which browser are you using?

**Review by kathyavini — APPROVED** [review]
Looks good!

@antsgar I am on Chrome as well. So I wonder if you have something different selected in MacOS itself -- @SayakaOno did some sleuthing once and found a system setting that controls scrollbar appearance in Chrome, but it's not a Chrome setting at all; kind of wild! Source: https://github.com/LiteFarmOrg/LiteFarm/pull/2794

Mine look pretty different depending on whether I have "Automatically based on mouse or trackpad" or "When scrolling" selected

**Review by SayakaOno — APPROVED** [review]
Looks good to me!

**Pull request merged**